specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx

```diff
@@ -306,8 +307,23 @@ function RecordSet<SCHEMA extends AnySchema>({
   }
 
   async function createNewRecordSet(
-    ids: RA<number | undefined>
+    ids: RA<number | undefined>,
+    fromBulkCarry: boolean = false
   ): Promise<void> {
+    if (
+      fromBulkCarry &&
+      typeof ids[0] === 'number' &&
+      typeof ids.at(-1) === 'number'
+    ) {
+      const startingResource = fetchResource('CollectionObject', ids[0]);
+      const endingResource = fetchResource('CollectionObject', ids.at(-1)!);
+      const startingResourceCatNumber = (await startingResource).catalogNumber;
+      const endingResourceCatNumber = (await endingResource).catalogNumber;
```

Comment:
once you change the code to remove handleCarryBulk and make handleAdd receive an array of resources, you won't need to do re-fetching of resources manually here as handleAdd is already being called with fetched resource